src/test/java/com/zufar/onlinestore/security/api/UserAuthenticationServiceTest.java

```diff
@@ -0,0 +1,110 @@
+package com.zufar.onlinestore.security.api;
+
+import com.zufar.onlinestore.security.dto.UserAuthenticationRequest;
+import com.zufar.onlinestore.security.jwt.JwtTokenProvider;
+import com.zufar.onlinestore.user.api.UserApi;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import java.util.Collection;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+
+@ExtendWith(MockitoExtension.class)
+class UserAuthenticationServiceTest {
+
+    @InjectMocks
+    private UserAuthenticationService userAuthenticationService;
+    @Mock
+    private JwtTokenProvider jwtTokenProvider;
+    @Mock
+    private AuthenticationManager authenticationManager;
+    @Mock
+    private UserApi userApi;
+    @Mock
+    private PasswordEncoder passwordEncoder;
+    @Mock
+    private Authentication authentication;
+    @Mock
+    private UserDetails userDetails;
+    private String hashPassword;
+    private UserAuthenticationRequest request;
+    private String jwtToken;
+
+    @BeforeEach
+    void setUp() {
+        request = new UserAuthenticationRequest("email", "password");
+        hashPassword = "password";
+        jwtToken = "jwtToken";
+
+        Mockito.when(userApi.getPasswordByEmail(any(String.class)))
+                .thenReturn(hashPassword);
+
+    }
+
+    @Test
+    @DisplayName("mock test authenticate")
+    void authenticate() {
```

Comment:
kinda bad naming for tests overally. Please take a look into bdd naming convention in tests